pyccel/parser/semantic.py

```diff
@@ -2069,6 +2062,7 @@ def _visit_For(self, expr, **settings):
             errors.report(INVALID_FOR_ITERABLE, symbol=expr.target,
                    bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
                    severity='error', blocker=self.blocking)
+        PyccelAstNode.stage = 'semantic'
```

Comment:
do you need to set the stage too semantic again here?  already done line 1996.

Reply:
1996 sets the stage to `syntactic`. This stage creates extra untreated nodes as they would appear after the syntactic stage. The only way I found to allow this to happen without it raising errors was to temporarily set the stage back to syntactic so the type checks are not carried out. This works as there are no `_visit` functions between line 1996 and here, but if you have a better solution that would a big improvement